features/include/pcl/features/good.h

```diff
@@ -0,0 +1,385 @@
+	* \param[in] sign either 1 or -1.
+	* \param[out] XOZ_histogram a 2D vector of unsigned int.
+        */            
+      void 
+      create2DHistogramFromXOZProjection ( PointCloudIn  XOZ_projected_view, double largest_side, 
+					    unsigned int number_of_bins, int sign, std::vector < std::vector<unsigned int> > &XOZ_histogram);
+      
+      /** \brief create a 2D histogram from XOY projection
+ 	* \param[in] XoY_projected_view pointer to a point cloud.
+	* \param[in] largest_side largest side of object bounding box.
+	* \param[in] number_of_bins  number of bins along one dimension.
+	* \param[in] sign either 1 or -1.
+	* \param[out] XOY_histogram a 2D vector of unsigned int.
+        */  
+      void 
+      create2DHistogramFromXOYProjection (PointCloudIn  XOY_projected_view,	double largest_side,
+					unsigned int number_of_bins, int sign, std::vector < std::vector<unsigned int> > &XOY_histogram);
+
+      /** \brief normalizing a 1D histogram
+ 	* \param[in] histogram 1D histogram (int).
+	* \param[out] normalized_histogram the resultant normalized_histogram (float).
+        */       
+      void
+      normalizingHistogram (std::vector <unsigned int> histogram, std::vector <float> &normalized_histogram);
+
+      /** \brief compute viewpoint entropy used for concatenating projections
+ 	* \param[in] normalized_histogram normalized_histogram (float).
+	* \param[out] entropy the resultant view entropy.
+        */   
+      void
+      viewpointEntropy (std::vector <float> normalized_histogram, float &entropy);
+
+      /** \brief find max view point entropy used for concatenating projections
+ 	* \param[in] view_point_entropy a vector of float contains three view entropies.
+	* \param[out] index the resultant projection index.
+        */  
+      void
+      findMaxViewPointEntropy (std::vector<float> view_point_entropy, int &index);
+
+      /** \brief compute average of view point histograms
+	* \param[in] histogram1 normalized histogram of projection 1.
+	* \param[in] histogram2 normalized histogram of projection 2
+	* \param[in] histogram3 normalized histogram of projection 3
+	* \param[out] average the resultant average of three histograms.
+        */  
+      void
+      averageHistograms (std::vector<float> histogram1, std::vector<float> historam2, std::vector<float> historam3, std::vector<float> &average);
+
+      /** \brief compute mean of a given histogram
+	* \param[in] histogram normalized histogram of projection.
+	* \param[out] mean the resultant mean.
+        */        
+      void
+      meanOfHistogram (std::vector<float> histogram, float &mean);
+
+      /** \brief compute variance of a given histogram
+	* \param[in] histogram normalized histogram of projection.
+	* \param[in] mean mean of the given histogram.
+	* \param[out] mean the resultant mean.
+        */        
+      void
+      varianceOfHistogram (std::vector<float> histogram, float mean, float &variance);

```

Comment:
You always use `meanOfHistogram` together with `varianceOfHistogram`. Why two separate functions? You can compute both mean and variance in a [single pass](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm).